restcomm/restcomm.telephony/src/main/java/org/restcomm/connect/telephony/CallManager.java

```diff
@@ -502,46 +527,59 @@ private void invite(final Object message) throws IOException, NumberParseExcepti
             logger.info("proxyIp: " + proxyIp);
         }
 
-        Client toClient = clients.getClient(toUser);
+        final Client toClient = clients.getClient(toUser);
 
         if (client != null) { // make sure the caller is a registered client and not some external SIP agent that we have little control over
             if (toClient != null) { // looks like its a p2p attempt between two valid registered clients, lets redirect to the b2bua
-                if(logger.isInfoEnabled()) {
-                    logger.info("Client is not null: " + client.getLogin() + " will try to proxy to client: "+ toClient);
-                }
-
-                ExtensionController ec = ExtensionController.getInstance();
-                IExtensionCreateCallRequest er = new CreateCall(fromUser, toUser, "", "", false, 0, CreateCallType.CLIENT, client.getAccountSid(), null,null, null, null);
-                ec.executePreOutboundAction(er, this.extensions);
+                long delay = sendPushNotificationIfNeeded(toClient.getPushClientIdentity());
```

Comment:
@agafox The Push Notification request should be sent AFTER the extension has been executed and when the call is allowed, that is after `if (er.isAllowed()) {`